### PR TITLE
AET-191 Fix e2e tests to be able to run it in CI

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?

Update e2e code to be able to run the test in CI:
- Fixed use of a private SSH key

### Motivation
Have new e2e tests running in the CI

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/new-e2e/utils/clients/ssh.go [hunk 1]
```diff
@@ -26,10 +26,10 @@ func GetSSHClient(user, host, privateKey string, retryInterval time.Duration, ma
 	return
 }
 
-func getSSHClient(user, host, privateKey string) (*ssh.Client, *ssh.Session, error) {
+func getSSHClient(user, host string, privateKey []byte) (*ssh.Client, *ssh.Session, error) {
 	var auth ssh.AuthMethod
 
-	if privateKey != "" {
+	if privateKey != nil {
 		privateKeyAuth, err := ssh.ParsePrivateKey([]byte(privateKey))
```
Comment (pgimalac) [inline test/new-e2e/utils/clients/ssh.go:33]: If `privateKey` is now a `[]byte` you can remove that conversion

[the remark this answers is not in the document]
Comment (KevinFairise2) [reply test/new-e2e/utils/clients/ssh.go:33]: Of course that's why I changed the type to []byte, but forgot to remove this. Nice catch!

File: test/new-e2e/utils/e2e/client/agent.go [hunk 2]
```diff
@@ -37,7 +40,21 @@ func NewAgent(installer *agent.Installer) *Agent {
...
+	}
+
+	if privateKeyPath != "" {
+		privateSshKey, err = os.ReadFile(privateKeyPath)
+		if err != nil {
+			return err
+		}
+	}
+
+	agent.vmClient, err = newVMClient(t, privateSshKey, &data.Connection)
```
[the remark this answers is not in the document]
Comment (pducolin) [reply test/new-e2e/utils/e2e/client/agent.go:57]: Checked code, `data.Connection` is imported from pulumi context, so makes sense to check store from the init function here, which the entry point of `UpResultDeserializer`

**Review by pducolin — APPROVED** [review]

File: test/new-e2e/utils/e2e/client/agent.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (KevinFairise2) [reply test/new-e2e/utils/e2e/client/agent.go:57]: I think we briefly talked about it with Olivier. But it was more straightforward like this, if in the future we need to pass more arguments to this function we could think about creating a context argument filled from the caller of 'initService'

File: test/new-e2e/utils/e2e/client/vm.go [hunk 3]
```diff
@@ -30,6 +33,20 @@ func NewVM(infraVM commonvm.VM) *VM {
 //lint:ignore U1000 Ignore unused function as this function is call using reflection
 func (vm *VM) initService(t *testing.T, data *commonvm.ClientData) error {
 	var err error
-	vm.vmClient, err = newVMClient(t, "", &data.Connection)
+	var privateSshKey []byte
```
Comment (ogaca-dd) [inline test/new-e2e/utils/e2e/client/vm.go:36]: This code is duplicated. It would be nice to factorize it!

**Pull request merged**